### PR TITLE
Enable using GPT timer as OS Timer on RT1xxx boards

### PR DESCRIPTION
**Pull request by danieldegrasse**
This switches the default kernel timer from Systick to the GPT timer on NXP RT1xxx boards. Switching from Systick to the GPT timer will allow RT boards to enter deeper sleep states without the kernel timer stopping. GPT1 is used as the hardware timer, and must be enabled in the board's DTS file with `status = "okay"` to enable the timer driver

### CODE REVIEW

**Review by mmahadevan108 — CHANGES_REQUESTED** [review]

File: drivers/counter/Kconfig.mcux_gpt [hunk 1]
```diff
@@ -7,4 +7,6 @@ config COUNTER_MCUX_GPT
 	bool "MCUX GPT driver"
 	depends on HAS_MCUX_GPT
 	help
-	  Enable support for mcux General Purpose Timer (GPT) driver.
+	  Enable support for mcux General Purpose Timer (GPT) driver. Note that
+	  enabling the mcux GPT as a counter will disable using it for the hardware
+	  clock, and the platform will fall back to using the systick timer
```
Comment (mmahadevan108) [inline drivers/counter/Kconfig.mcux_gpt:12]: `, and the platform will fall back to using the systick timer` I think this should be deleted as the current design may not have this.

File: drivers/timer/Kconfig [hunk 2]
```diff
@@ -417,6 +417,18 @@ config MCUX_OS_TIMER
 	  This module implements a kernel device driver for the NXP OS
 	  event timer and provides the standard "system clock driver" interfaces.
 
+DT_NODELABEL_NXP_GPT_TIMER := gpt2
+
```
Comment (mmahadevan108) [inline drivers/timer/Kconfig:421]: This is limiting to `gpt2`, we should try to keep this generic in case future platforms have a different gpt

**Review by mmahadevan108 — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by danieldegrasse** [comment]
Adding DNM, as some tests are still failing on this PR that CI has not caught.

**Comment by danieldegrasse** [comment]
PR has been update to fix CI failures. One of the CI failures was due to a kernel test making an incorrect assertion, and I've added a change to this PR to fix it. 

`tests/kernel/threads/thread_apis` checks `k_busy_wait` by delaying for 100us, then verifying the thread usage time has increased by at least 100us worth of hardware cycles. 

However, the assertion used the macro `zassert_true(dt >= k_cyc_to_us_floor64(100), NULL);`, which is incorrect: this is comparing the elapsed cycles to the number of microseconds that pass in 100 cycles. For most fast hardware clocks, this assertion will pass, since very few microseconds pass in 100 cycles. However, for the 32KHz GPT clock this assertion will fail. The correct assertion is the following: `zassert_true(dt >= k_us_to_cyc_floor64(100), NULL);`

**Comment by danieldegrasse** [comment]
CI is failing because of a bug introduced by #40183, #41025 fixes the issue

**Comment by danieldegrasse** [comment]
Removing DNM label, as PR is now passing all CI tests internally

**Pull request merged**